src/tracer.js

```diff
@@ -254,23 +259,20 @@ export default class Tracer {
     }
 
     /**
-     *  Makes a concrete sampling decision for the ctx span context based on information
-     *  available to it's child span.
-     *
-     *  The ctx context's deferred mask is unset after this decision is made
+     * Reapplies sampling if the parent span isn't sampled
      *
+     * @param parentCtx the parent span context
      * @param ctx the span context
      * @param operationName the operation name for the child
      * @param tags tags to be applied by the sampler
      */
-    processDeferredSampling(ctx: SpanContext, operationName: string, tags: any) {
-        if (ctx.isDeferredSampling) {
+    upsampleChild(parentCtx: SpanContext, ctx: SpanContext, operationName: string, tags: any) {
+        if (!parentCtx.isSampled() && !parentCtx.isDebug()) {
             if (this._sampler.isSampled(operationName, tags)) {
                 ctx._flags |= constants.SAMPLED_MASK;
             } else {
                 ctx._flags &= ~constants.SAMPLED_MASK;
```

Comment:
> from when a span is created to when the span is injected/finished/used as a parent

doesn't it mean instead of doing it in one place we're going to do it in 3 places?